src/App.tsx

```diff
@@ -86,34 +85,44 @@ const SearchText = styled.h2`
   }
 `;
 
+const locationSearch = () => {
+  return window.location.search.startsWith('?q=');
```

Comment:
any chance that search is null here? `window.location.search?.startsWith('?q=')`?